libs/core/src/lib/resizable-card-layout/resizable-card-item/resizable-card-item.component.ts

```diff
@@ -0,0 +1,484 @@
+        this._resizeDebounce$.complete();
+    }
+
+    /**
+     * When resize handler is pressed and resizing may start.
+     * @param event: MouseEvent
+     * @param resizeDirection: which handler is pressed to resize
+     */
+    onMouseDown(event: MouseEvent, resizeDirection: ResizeDirection): void {
+        event.preventDefault();
+        if (!this.resizable) {
+            return;
+        }
+
+        this.showBorder = true;
+        this.showingResizeIcon = true;
+        this._resize = true;
+        this._prevX = event.clientX;
+        this._prevY = event.clientY;
+        this._prevCardWidth = this.cardWidth;
+        this._prevCardHeight = this.cardHeight;
+        this._resizeDirection = resizeDirection;
+    }
+
+    /**
+     * When mouse moves to resize the card.
+     * using window:mousemove so, resize will happen smoothly
+     * @param event: MouseEvent
+     */
+    @HostListener('window:mousemove', ['$event'])
+    onMouseMove(event: MouseEvent): void {
+        event.preventDefault();
+        if (!this.resizable) {
+            return;
+        }
+
+        if (!this._resize) {
+            return;
+        }
+
+        // resizing card will go over other cards
+        this.zIndex = 1;
+
+        if (this._resizeDirection === 'both') {
+            this._horizontalResizing(event.clientX);
+            this._verticalResizing(event.clientY);
+        } else if (this._resizeDirection === 'horizontal') {
+            this._horizontalResizing(event.clientX);
+        } else {
+            this._verticalResizing(event.clientY);
+        }
+
+        // Emit resizing event on some interval. improves performance.
+        this._resizeDebounce$.next(new PositionChange(this._prevX - event.clientX, this._prevY - event.clientY));
+
+        const heightDiff = Math.abs(this.cardHeight - this._prevCardHeight);
+        const widthDiff = Math.abs(this.cardWidth - this._prevCardWidth);
+
+        // if card dimension change in step. horizontal or vertical.
+        if (
+            (heightDiff > 0 && heightDiff % verticalResizeStep === 0) ||
+            (widthDiff > 0 && widthDiff % HorizontalResizeStep === 0)
+        ) {
+            this.stepChange.emit(this._getResizedEventObject());
+        }
+
+        this._prevX = event.clientX;
+        this._prevY = event.clientY;
+    }
+
+    /**
+     * when resizing of card stops
+     * @param event: MouseEvent
+     */
+    @HostListener('document: mouseup', ['$event'])
+    onMouseUp(event: MouseEvent): void {
+        if (!this.resizable) {
+            return;
+        }
+
+        // increase/decrease width of card in order of 20rem
+        if (Math.abs(this.cardWidth - this._prevCardWidth) > 0) {
+            this._horizontalStepResizing();
+        }
+
+        // increase/decrease height of card in order of 1rem
+        if (Math.abs(this.cardHeight - this._prevCardHeight) > 0) {
+            this._verticalStepResizing();
+        }
```

Comment:
Could you do some offset, just to not let user resize cards by default, for example user needs to resize by 20px to change width/height?

![Peek 2021-03-23 12-04](https://user-images.githubusercontent.com/26483208/112136930-ed42c800-8bcf-11eb-87e0-dec9140f3e9f.gif)

Reply:
added horizontalOffset of 20px and verticalOffset of 10px. now it works fine.